File: Client/job-khojo/src/app/core/models/job.model.ts
import { SafeHtml } from '@angular/platform-browser';

export class Job {
  jobId!: number;
  title!: string;
  subtitle!: string;
  description!: string;
  openings!: number;
  salary!: number;
  location!: number;
  jobType!: number;
  createdBy!: number;
  isActive!: boolean;
  appliedBy?: number;
  isSaved?: boolean;
}

File: Data/Models/JobModel.cs
using System.ComponentModel.DataAnnotations;
using Data.Validations;

namespace Data.Models;

public class JobModel
{
    public int JobId { get; set; }

    [Required(ErrorMessage = "Title is required"), WhiteSpaceValidator]
    public string Title { get; set; }

    public string Subtitle { get; set; }

    [Required(ErrorMessage = "Description is required"), WhiteSpaceValidator]
    public string Description { get; set; }

    [Required(ErrorMessage = "No. of Openings are required"), WhiteSpaceValidator]
    public int? Openings { get; set; }

    public int? Salary { get; set; }

    public int? JobType { get; set; }

    public int? Location { get; set; }

    public int CreatedBy { get; set; }

    public bool? IsActive { get; set; }

    public int AppliedBy { get; set; }

    public bool isApplied { get; set; }
}
